api/proto/teleport/legacy/types/types.proto

```diff
@@ -1012,7 +1012,11 @@ message IdentityCenterPermissionSet {
 
   // AssignmentID is the ID of the Teelport Account Assignment resource that
   // represents this permission being assigned on the enclosing Account.
-  string AssignmentID = 3 [(gogoproto.jsontag) = "assignment_name,omitempty"];
+  string AssignmentID = 3 [(gogoproto.jsontag) = "assignment_id,omitempty"];
+
+  // RequiresRequest indicates that the user needs to raise an access request to
+  // be granted this permission set
+  bool RequiresRequest = 4 [(gogoproto.jsontag) = "assignment_name,omitempty"];
```

Comment:
Why is the RequiresRequest attribute needed for the permission_set object?

The RequiresRequest attribute is only used by the Front End.  For the same flow in Okta, when an app has multiple okta_groups or AWS app access with multiple IAM configurations, the RequiresRequest attribute was not necessary apart from the top level app property and Instead, the logic riels on the top-level app.RequiresRequest flag. 

So why is it handled differently in this case? For the AWS IC flow, why is the RequiresRequest flag needed for each selected object, whereas for other flows (e.g., Okta with multiple okta_groups), the logic relies on the top-level app.RequiresRequest flag?


Also does the FE is aware of this change ?
 cc: @flyinghermit

Reply:
We could probably do this with the enriched resource, outside of the `AppV3` and avoiding the protobuf change.

Reply:
OK, so the reason this needs to be in the protobuf definition is that the `app` representing the IC Account goes over the wire (at least logically) at some point between the Unified Resource Cache listing and being rendered to json.

As far as I can tell, we _need_ to calculate which Account Assignments are available to (or requestable by) a user _during_ this listing, because once we have transmitted the resource to the URC client, we no longer have the tools available to evaluate the user's access. 

If we want to expose this requestability info to the UI, then we need to expose it to the Unified Resource Cache client _first_, and for that it has to be part of the App's wire format.